src/tests/modules/bson.rs

```diff
@@ -306,3 +307,18 @@ fn system_time() {
         0
     );
 }
+
+#[test]
+fn debug_print() {
+    #[derive(Debug)]
+    struct Example {
+        a: i32,
+        b: String,
+    }
+
+    let doc = doc! { "a": 1, "b": "data" };
+    let data = indexmap! {"a" => Bson::from(1), "b" => Bson::from("data")};
```

Comment:
Rather than testing against the internal representation here, we should define a String for what we want this to look like. That way, if we ever change the internal representation of `Document`, we can maintain a consistent debug format, and we also won't have to update this test.

Reply:
Makes sense. Is there a possibility of the external representation changing? Or is that just not very likely/not possible?

Reply:
The likelihood is low, though we did do it before (we used to use `linked-hash-map` instead of `indexmap`). Another benefit of defining the expected result absolutely (i.e. as a string) rather than relative to `indexmap` is that it also protects us from the situation where `indexmap` changes its own `Debug` implementation to something that differs from the one we want / currently use. More abstractly put, by comparing to `indexmap`'s output, the test verifies only the implementation, but by comparing to a pre-defined string, it also tests the stability of the public interface.